### PR TITLE
[Bundles] Update best practices to use Symfony Flex and do not favor any CI tool

### PR DESCRIPTION
**Pull request by wouterj**
Fixes https://github.com/symfony/symfony-docs/issues/10361

I think it's better to not show a complete Travis CI config, but instead document the intentions behind it. This way, everyone can implement them on their favorite CI tool.

Meanwhile, I removed the `symfony/lts` usage instead of the now recommended `SYMFONY_REQUIRE` env var.

### CODE REVIEW

File: bundles/best_practices.rst [hunk 1]
```diff
@@ -171,73 +171,59 @@ Continuous Integration
...
+and `Travis CI`_.
+
+A bundle should at least test:
+
+* The lower bound of their dependencies (by running ``composer update --prefer-lowest``);
+* The supported PHP versions;
+* All supported major Symfony versions (e.g. both ``3.x`` and ``4.x`` if
+  support is claimed for both).
+
+Thus, a bundle support PHP 7.3, 7.4 and 8.0, and Symfony 3.4 and 4.x should
```
Comment (noniagriconomie) [inline bundles/best_practices.rst:184]: Is it still worth documenting v3.4?

Also for the travis config file, is it possible to use the langage toggle config feature php/xml/yaml but instead use it for toggling between for example github actions, travis, etc ?

Comment (dbu) [reply bundles/best_practices.rst:184]:
```suggestion
Thus, a bundle supporting PHP 7.3, 7.4 and 8.0, and Symfony 3.4, 4.x and 5.x should
```
listing explicit versions is prone to be outdated after a while, but i think this is about the example, so getting outdated is ok.

File: bundles/best_practices.rst [hunk 2]
```diff
@@ -171,73 +171,59 @@ Continuous Integration
...
-Consider using the `Travis cron`_ tool to make sure your project is built even if
-there are no new pull requests or commits.
+providing this feature for free for open source projects, like `GitHub Actions`_
+and `Travis CI`_.
+
+A bundle should at least test:
+
+* The lower bound of their dependencies (by running ``composer update --prefer-lowest``);
+* The supported PHP versions;
+* All supported major Symfony versions (e.g. both ``3.x`` and ``4.x`` if
```
Comment (dbu) [inline bundles/best_practices.rst:181]:
```suggestion
* All supported major Symfony versions (e.g. both ``4.x`` and ``5.x`` if
```

**Review by dbu — COMMENTED** [review]
thanks for the update. i find it a pitty to completely drop the example config file, it is quite nice to see a full example (also great to quickstart a repo). can we preserve that file somehow somewhere? i was thinking of putting it in a gist, but that is a bit fragile and tied to a personal account. linking a concrete example in a popular OSS bundle is also prone to not work at some point. could we have a non-rst file in the doc that can be downloaded?

**Comment by javiereguiluz** [comment]
I disagree with this 😐   I'd love to provide a realistic example that readers can copy+paste to test their bundles.

**Comment by wouterj** [comment]
Alright, what do we do then? Given [Travis changes its paid plan](https://twitter.com/lsmith/status/1323514135791939584), does it makes sense to replace the Travis config with GitHub actions? (as it'll probably be a more common use-case for new bundles starting now)

Or as, suggested by @noniagriconomie, do we want tabs? The "issue" with tabs is that we need to explicitly add them to the Sphinx config.

**Comment by wouterj** [comment]
Something else I was just thinking: what about creating a template repository for new bundles? That would include the README, GitHub actions config, main directories, etc.

**Comment by javiereguiluz** [comment]
That could be a nice idea. I feel like the "onboarding" for bundle authors has too much friction. We could have a `make:bundle` maker, a template repository, better docs about how to create a bundle and support multiple Symfony versions, how to create a "fake" Symfony app to functional test the bundle, etc.

**Comment by javiereguiluz** [comment]
Thank you Wouter!

**Pull request merged**